api/src/main/java/io/opentelemetry/common/Attributes.java

```diff
@@ -199,24 +201,7 @@ public Attributes build() {
 
     /** Sets a {@link AttributeKey} with associated value into this. */
     public <T> Builder setAttribute(AttributeKey<T> key, T value) {
-      if (key == null || key.getKey() == null || key.getKey().length() == 0) {
-        return this;
-      }
-      if (value == null) {
-        // Remove key/value pairs
-        Iterator<Object> itr = data.iterator();
-        while (itr.hasNext()) {
-          AttributeKey k = (AttributeKey) itr.next();
-          if (key.equals(k)) {
-            // delete key and value
-            itr.remove();
-            itr.next();
-            itr.remove();
-          } else {
-            // skip the value part
-            itr.next();
-          }
-        }
+      if (value == null || key == null || key.getKey() == null || key.getKey().length() == 0) {
```

Comment:
Nit: for simmetry with the other checks
```suggestion
      if (key == null || key.getKey() == null || key.getKey().length() == 0 || value == null ) {
```